### PR TITLE
[Tech] Add ts-prune to find dead code

### PR DESCRIPTION
**Pull request by Nocccer**
This PR adds ts-prune to find dead code or useless exports. 

- All dead code reported by ts-prune were removed
- Adapt tests to still test the old now not anymore exported functions
- Moved function notify to dialog. I didn't find a way to mock it from utils while testing utils functions. Utils is a large file and making it smaller makes sense anyway

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [x] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/backend/legendary/games.ts [hunk 1]
```diff
@@ -857,19 +857,11 @@ class LegendaryGame extends Game {
       "Legendary's config from config.ini (before Heroic's settings):\n"
     )
 
-    try {
-      const json = JSON.parse(stdout)
-      // remove egl auth info
-      delete json['egl_parameters']
+    const json = JSON.parse(stdout)
```
Comment (arielj) [inline src/backend/legendary/games.ts:860]: we shouldn't remove this try/catch

File: src/backend/dialog/dialog.ts [hunk 2]
```diff
@@ -52,4 +53,22 @@ function showDialogBoxModalAuto(props: {
...
 
-export { showDialogBoxModalAuto }
+type NotifyType = {
+  title: string
+  body: string
+}
+
+function notify({ body, title }: NotifyType) {
+  if (Notification.isSupported() && !isSteamDeckGameMode) {
+    const mainWindow = BrowserWindow.getAllWindows()[0]
```
Comment (arielj) [inline src/backend/dialog/dialog.ts:63]: we can use `getMainWindow` from the `main_window` module here to hide the implementation

File: src/backend/sideload/games.ts [hunk 3]
```diff
@@ -240,42 +229,6 @@ export async function launchApp(appName: string): Promise<boolean> {
   return false
 }
 
-export async function moveInstall(
```
Comment (arielj) [inline src/backend/sideload/games.ts:243]: I wonder if this is ok to be removed or if it means that at some point we lost the code to trigger the move of the installation for sideloaded apps?

cc @flavioislima to confirm

Comment (Nocccer) [reply src/backend/sideload/games.ts:243]: Got fixed by merging with newest commits.

Comment (Nocccer) [reply src/backend/sideload/games.ts:243]: Ok vscode github pull request extension tricked me. If i click on your comment it jumps to this function and shows me that it is added?

Comment (Nocccer) [reply src/backend/sideload/games.ts:243]: @flavioislima do we need the moveInstall function for sideload games?

File: .github/workflows/codecheck.yml [hunk 4]
```diff
@@ -17,3 +17,5 @@ jobs:
         run: yarn
       - name: Check Typescript syntax
         run: yarn codecheck
+      - name: Find dead code
+        run: yarn find-deadcode
```
Comment (CommandMC) [inline .github/workflows/codecheck.yml:21]: *Maybe* do this in a separate Workflow?

**Review by CommandMC — APPROVED** [review]

File: .github/workflows/codecheck.yml (hunk 4 above)
Comment (Nocccer) [reply .github/workflows/codecheck.yml:21]: We should avoid to make a new workflow for each small checker.
I think the limits are 6 workflows in parallel. We would waste time here to setup a ubuntu server and install dependencies.

Comment (Nocccer) [reply .github/workflows/codecheck.yml:21]: Ok 20 is the limit, still i would not cluster the checker with to much workflows.

Comment (Nocccer) [reply .github/workflows/codecheck.yml:21]: Added to linter workflow

File: src/backend/online_monitor.ts [hunk 5]
```diff
@@ -119,11 +119,11 @@ export const initOnlineMonitor = () => {
   })
 }
 
-export const makeNetworkRequest = (callback: () => unknown) => {
-  if (isOnline()) {
-    callback()
-  }
-}
+// export const makeNetworkRequest = (callback: () => unknown) => {
```
Comment (flavioislima) [inline src/backend/online_monitor.ts:122]: Why to keep this commented code?

**Pull request merged**